adapters/yieldlab/yieldlab.go

```diff
@@ -225,22 +225,24 @@ func (a *YieldlabAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externa
 			}
 		}
 
+		imp := a.findImp(bid.ID, internalRequest.Imp)
```

Comment:
The fact that the `a.findImp(bid.ID, internalRequest.Imp)` call traverses all of the imps in `internalRequest.Imp` at every loop makes `MakeBids` to have a time complexity of `O(nm)` where `n` is the number of bids and `m` the number of imps inside `internalRequest.Imp`. If we expect bids to come with unique `bid.ID`'s maybe we could simplify to `O(n+m)` at the expense of some memory by creating a map. (Prebid server has some memory to spare compared to CPU time). Do you think something like this would be a godd idea?
```
210     bidderResponse := &adapters.BidderResponse{
211         Currency: currency.EUR.String(),
212         Bids:     []*adapters.TypedBid{},
213     }
214
    +   // At a expense of O(m) memory
    +   var adslotToImpMap map[string]*openrtb2.Imp
    +   for i:=0; i<len(internalRequest.Imp); i++ {
    +       // extract adslotID from Ext
    +       adslotID := someFunctionToExtractAdslotID(internalRequest.Imp[i])
    +       if internalRequest.Imp[i].Video != nil || internalRequest.Imp[i].Banner != nil {
    +           // we could only allow Video or Banners into the map 
    +           adslotToImpMap[adslotID] = &internalRequest.Imp[i]
    +       }
    +   }
    + 
215     for _, bid := range bids {
216         width, height, err := splitSize(bid.Adsize)
217         if err != nil {
218             return nil, []error{err}
219         }
220
221         req := a.findBidReq(bid.ID, params)
222         if req == nil {
223             return nil, []error{
224                 fmt.Errorf("failed to find yieldlab request for adslotID %v. This is most likely a programming issue", bid.ID),
225             }
226         }
227
228  -      imp := a.findImp(bid.ID, internalRequest.Imp)
229
     +      // Does `bid.ID` match with any element in our map?
     +      if imp, exists := goMap[bid.ID]; !exists {
     +          continue
     +      } else {
     +          // Imp matches, do all the rest
                 . 
                 . 
                 . 
 adapters/yieldlab/yieldlab.go
```
Unless of course none of this makes sense in your context. Maybe you know you expect responses to only come with one bid, or imp arrays with only one imp. You decide I'm good either way.

Reply:
This is a very good one - thanks again not only for bringing this up, but in particular for giving very helpful pointers in how this could be implemented! I wasn't particularly happy with our original solution but as I have very little experience with Go so far, it was all I could come up with. I have just updated this PR with another commit which implements the suggested solution from above. Let me know if that looks good to you!